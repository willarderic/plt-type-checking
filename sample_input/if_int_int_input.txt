(if 0 1 (input))
