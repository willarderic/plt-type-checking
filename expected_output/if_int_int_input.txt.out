TOKEN: LPAREN    LEXEME: (
TOKEN: IF        LEXEME: if
TOKEN: INTCONST  LEXEME: 0
CONST -> INTCONST
EXPR -> CONST
TOKEN: INTCONST  LEXEME: 1
CONST -> INTCONST
EXPR -> CONST
TOKEN: LPAREN    LEXEME: (
TOKEN: INPUT     LEXEME: input
INPUT_EXPR -> input
PARENTHESIZED_EXPR -> INPUT_EXPR
TOKEN: RPAREN    LEXEME: )
EXPR -> ( PARENTHESIZED_EXPR )
IF_EXPR -> if EXPR EXPR EXPR
PARENTHESIZED_EXPR -> IF_EXPR
TOKEN: RPAREN    LEXEME: )
EXPR -> ( PARENTHESIZED_EXPR )
START -> EXPR

---- Completed parsing ----

